include/libtorrent/alert_types.hpp

```diff
@@ -2355,12 +2355,27 @@ namespace libtorrent
 		int m_size;
 	};
 
+	struct TORRENT_EXPORT dht_get_peers_reply_alert: alert {
+
+		dht_get_peers_reply_alert(aux::stack_allocator& alloc
+			, sha1_hash const& ih
+			, std::vector<tcp::endpoint> const& v);
+
+		const static int static_category = alert::dht_notification;
+		TORRENT_DEFINE_ALERT(dht_get_peers_reply_alert, 87)
+
+		virtual std::string message() const;
+
+		sha1_hash info_hash;
+		std::vector<tcp::endpoint> peers;
```

Comment:
see: https://github.com/arvidn/libtorrent/blob/master/src/alert.cpp#L269

I believe all alerts using the stack_allocator use it to copy strings. to hold tcp::endpoint objects, you'd have to use a different allocation function, see: https://github.com/arvidn/libtorrent/blob/master/include/libtorrent/stack_allocator.hpp#L70

one issue is that the allocator only provides raw bytes. It can only hold POD types, since it may have to grow. It wouldn't make the API quite as elegant (basically it could hold a compact representation of IPv4 + port and a separate array of IPv6 + port)